tests/integration/molecule/module_secret/converge.yml

```
@@ -0,0 +1,173 @@
+---
+- name: Converge
+  hosts: all
+  gather_facts: false
+
+  tasks:
+    - name: Make sure we start clean
+      sensu.sensu_go.secret_info:
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is success
+          - result.objects | length == 0
+
+    - name: Create new secret (check mode)
+      sensu.sensu_go.secret: &secret
+        name: sample-secret
+        provider: env
+        id: MY_ENV_VAR
+      check_mode: true
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+          - "result.object.metadata.name == 'sample-secret'"
+          - "result.object.id == 'MY_ENV_VAR'"
+          - "result.object.provider == 'env'"
+
+    - name: Make sure things are still clean
+      sensu.sensu_go.secret_info:
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.objects | length == 0
+
+    - name: Create new secret
+      sensu.sensu_go.secret: *secret
+      register: result
+    - ansible.builtin.assert:
```

```suggestion
    - ansible.builtin.assert: *create-assertions
```
If you decide to go with `&create-assertions` anchor (see my earlier comment).

Personally, it makes more sense to me to either use aliases for both module params + assertions, or for neither. To me, seeing the assertions but not module params means more mental overhead, because I have to scroll back to the `&secret`, inspect what's there, and re-check that the assertions make sense.